### ISSUE TITLE
BTRFS not working in Virt-Manager

### ISSUE THREAD

**Issue by AnhVo1989** [archlinux/archinstall#1245]
I'm testing out the latest version of archinstall in Virt-Manager (version 2.4.3rc1) by git cloning the archinstall repo while in the live environment and have noticed that it doesn't seem to like BTRFS. EXT4 works as expected. 

It gets to mounting the partitions, hits some kind of bug, and spits out a bunch of text that ends with "TypeError: cannot pickle 'generator' object"

I've only been testing through Virt-Manager so I dont know if 2.4.3rc1 acts any differently on bare metal. 

Anyone encounter this? Any guidance is appreciated

Originally posted on Reddit - https://www.reddit.com/r/archlinux/comments/uxetf4/archinstall_btrfs_not_working_on_virtmanager/?utm_medium=android_app&utm_source=share

**Comment by dylanmtaylor** [comment]
Can you try on the master development iso? @Torxed just pushed a commit that probably fixes this

**Comment by Anh-Linux** [comment]
No luck on my end! The msg indicating where to find the error log was really helpful though! Tried installing on Virt-Manager using 2.4.3rc1 as of 6:50pm EST. Here is my  install log

[install.log](https://github.com/archlinux/archinstall/files/8782632/install.log)

**Comment by Torxed** [comment]
You would need to try with `master` instead of RC1 because i havent been able to tag a new version yet.

**Comment by Anh-Linux** [comment]
How to I do that?

**Comment by Anh-Linux** [comment]
Or is that the version included in the May ISO?

**Comment by dylanmtaylor** [comment]
Every time a commit is made against master (https://github.com/archlinux/archinstall/commits/master) an ISO is generated. Click the check mark, click details near Build Arch ISO with ArchInstall Commit, click summary. For example, https://github.com/archlinux/archinstall/actions/runs/2392251445. Then click the "[Arch Live ISO](https://github.com/archlinux/archinstall/suites/6675421827/artifacts/252908790)" link.

**Comment by dylanmtaylor** [comment]
There's also a list of ISOs built at https://github.com/archlinux/archinstall/actions/workflows/iso-build.yaml

**Comment by Anh-Linux** [comment]
Thank you so much! So after downloading the master development isos, i would just run archinstall as normal?

**Comment by dylanmtaylor** [comment]
Exactly. It's a special Arch Linux install build with the EXACT commit that you download the ISO for built-in. You just run it as usual. Fair warning, it might have bugs, as it is unreleased software. ;)

**Comment by Anh-Linux** [comment]
Here is the install log I pulled after trying two installations with the Master ISO. 1 unsuccessful run and 1 successful run.

TL;DR : The default subvolume layout when using BTRFS (no encryption) works as intended but adding @root mounted at /root did not work. So I don't know if this issue should be marked closed?

[install.log](https://github.com/archlinux/archinstall/files/8783347/install.log)

**Comment by Anh-Linux** [comment]
v2.4.2 - @root mounted at /root can be add to the disk layout json file and the installer runs. Manually creating the subvol and mount point causes the installer to crash.
|
|
v2.4.3rc1 - btrfs causes the installer to crash regardless of whether ran with or with declarative json file.
|
|
Master ISO - Creating @root mounted at /root using the installer causes it to crash. I haven't tried tried from a declarative file yet. I will try that asap and report back.

^^ Hope this helps!

**Comment by AnhVo1989** [comment]
3-run install log and Konsole output from the third install.

1 unsuccessful and 2 successful runs

[install.log](https://github.com/archlinux/archinstall/files/8783796/install.log)

[vm-konsole-output.log](https://github.com/archlinux/archinstall/files/8783794/vm-konsole-output.log)

**Comment by Torxed** [comment]
One thing stands out that is a bit strange:
```
Could not retrieve subvolume information from /mnt/archinstall/@root: ['/usr/bin/btrfs', 'subvolume', 'show', '/mnt/archinstall/@root'] exited with abnormal exit code [256]: b"ERROR: cannot find real path for '/mnt/archinstall/@root': No such file or directory\r\n"
```
I'm not sure why it's trying to call `btrfs subvolume show /mnt/archinstall/@root`, it should be `btrfs subvolume show /mnt/archinstall/root`.

I'll investigate this today.

**Comment by Torxed** [comment]
I think this should be fixed now, you can try the ISO from here once the runner is done: https://github.com/archlinux/archinstall/actions/runs/2401200385

**Comment by AnhVo1989** [comment]
Ill download and run a quick install in a second! Thanks guys!!!

**Comment by Torxed** [comment]
If it still doesn't work, drop a comment here and I'll re-open the ticket :)

**Comment by AnhVo1989** [comment]
On the  ISO on the link you sent (the latest master?)

Trying to edit the default subvolumes gave me this

![image](https://user-images.githubusercontent.com/104331316/170842275-5196570a-1d2c-49d5-8dbb-3a40584ff467.png)

**Comment by AnhVo1989** [comment]
Editing user-mader subvol results in the same

![image](https://user-images.githubusercontent.com/104331316/170842416-0f1a3236-0073-4c66-80a2-4f9ed77ea11c.png)

**Comment by Torxed** [comment]
@svartkanin You're the menu wizard, would this be easy to solve in time for a release today, or we live with this and take it for the next release? :)

Since users can edit `user_disk_layout.json`.

**Comment by svartkanin** [comment]
@Torxed I'll have a look at it, not sure if I can find it right away :)

**Comment by Torxed** [comment]
I'll give it a few hours :) No need to rush it, but if it's an easy fix that would be awesome :D

**Comment by svartkanin** [comment]
@Torxed here's a quick one https://github.com/archlinux/archinstall/pull/1277

This will prevent the blowup, but I saw that the subvolume menu is quite broken in general so that will need some reworking for sure. I don't have the time to get something out today but will look into it in the next week or so. 
From what I saw quickly, when editing an added subvolume the menus don't work properly (can't edit name for example) and cancel doesn't really cancel but also save I think. Not a big deal but this needs to be fixed.
